tests/ftest.sh

```diff
@@ -14,13 +14,115 @@ VIRTUAL_ENV="$ROOT_DIR/.venv"
+# This is needed for the release process when the ES snapshot image may not yet be available.
+# Snapshot images are pushed to the registry by the unified release workflow.
+
+# Function to resolve the DRA manifest URL
+function resolve_dra_manifest {
+  DRA_ARTIFACT=$1
+  DRA_VERSION=$2
+
+  # Perform the curl request and capture the output and exit code
+  RESPONSE=$(curl -sS -f $ARTIFACT_BASE_URL/$DRA_ARTIFACT/latest/$DRA_VERSION.json 2>&1)
+  CURL_EXIT_CODE=$?
+
+  # Check if the curl command failed
+  if [ $CURL_EXIT_CODE -ne 0 ]; then
+    echo "Error: Failed to fetch DRA manifest for artifact $DRA_ARTIFACT and version $DRA_VERSION."
+    echo "Details: $RESPONSE"
+    exit 1
+  fi
+
+  # Extract the manifest_url using jq
+  MANIFEST_URL=$(echo "$RESPONSE" | jq -r '.manifest_url' 2>/dev/null)
+
+  # Check if the jq command succeeded and if manifest_url is non-empty
+  if [ -z "$MANIFEST_URL" ] || [ "$MANIFEST_URL" == "null" ]; then
+    echo "Error: No manifest_url found in the response for artifact $DRA_ARTIFACT and version $DRA_VERSION."
+    echo "Response: $RESPONSE"
+    exit 1
+  fi
+
+  # Output the manifest URL
+  echo "$MANIFEST_URL"
+}
+
+# Function to download the Docker image tarball
+function download_docker_tarball {
+  TAR_URL=$1
+  TAR_FILE=$2
+
+  echo "Downloading Docker image tarball from $TAR_URL..."
+  curl -O "$TAR_URL"
+
+  if [ ! -f "$TAR_FILE" ]; then
+    echo "Error: Download failed. File $TAR_FILE not found."
+    exit 1
+  fi
+}
+
+# Function to load the Docker image directly from the tarball
+function load_docker_image {
+  TAR_FILE=$1
+  echo "Loading Docker image from $TAR_FILE..."
+  docker load < "$TAR_FILE"
+}
+
+# Determine system architecture
+ARCH=$(uname -m)
+
+# Normalize architecture name
+if [[ $ARCH == "arm64" ]]; then
+  ARCH="aarch64"
+fi
+
+# Select the appropriate Docker tarball name based on architecture
+case $ARCH in
+  x86_64)
+    DOCKER_TARBALL_NAME="elasticsearch-$VERSION-docker-image.tar.gz"
+    ;;
+  aarch64)
+    DOCKER_TARBALL_NAME="elasticsearch-$VERSION-docker-image-aarch64.tar.gz"
+    ;;
+  *)
+    echo "Error: Unsupported architecture $ARCH"
+    exit 1
+    ;;
+esac
+
+# Get the DRA manifest URL for Elasticsearch
+ELASTICSEARCH_DRA_MANIFEST=$(resolve_dra_manifest "elasticsearch" $VERSION)
+
+# Parse Docker image tarball information
+DOCKER_TARBALL_URL=$(curl -sS "$ELASTICSEARCH_DRA_MANIFEST" | jq -r ".projects.elasticsearch.packages.\"$DOCKER_TARBALL_NAME\".url")
+
+if [ -z "$DOCKER_TARBALL_URL" ] || [ "$DOCKER_TARBALL_URL" == "null" ]; then
+  echo "Error: Docker tarball URL not found in the manifest."
+  exit 1
+fi
+
+# Execute the functions to download and load the image
+download_docker_tarball "$DOCKER_TARBALL_URL" "$DOCKER_TARBALL_NAME"
+load_docker_image "$DOCKER_TARBALL_NAME"
+
+# Export image name following DRA conventions
+export ELASTICSEARCH_DRA_DOCKER_IMAGE="elasticsearch:$ARCH"
```

Comment:
How well would that work if you run ftests locally on different branches (9.0 and 8.x) for example?

Since there's no version it can have weird conflicts, right? Or is it resolved somewhere else?

Reply:
It works well, I tried it with e.g. 8.16 

We are fetching the manifest for the right version that depends on the connector version (in the VERSION file) see this line https://github.com/elastic/connectors/pull/2979/files#diff-2c88fda54d210cfb99f46dd547063eb3b88f8f41296ce60881535313cd7b281cR105 and you can backtrack where the version variable comes from. 

Unfortunately the `elasticsearch` images loaded from tarball do not have the "stack version" tag, instead they are just in format  `elasticsearch:{aarch60|x86_64}`, so the arch is the tag.

But given that we are fetching it from the right manifest URL it is the correct version.

Reply:
E.g. this is for SPO ftests in 9.0.0
<img width="661" alt="Screenshot 2024-11-19 at 13 07 13" src="https://github.com/user-attachments/assets/3b45b46a-9fbc-45cc-8b71-61a146e36f90">

Reply:
If you run first on `main` branch, then checkout to 8.16 and run ftest again, the docker load will override the the image for your arch and the logs will show
```
[FMWK][12:10:39][INFO] Elasticsearch 8.16.0-SNAPSHOT and Connectors 8.16.0 are compatible
```
( I tried this sequence of steps locally by tweaking the version in the file)